pkg/resource/utils.go

```diff
@@ -0,0 +1,34 @@
+package resource
+
+import (
+	"context"
+
+	"go.uber.org/atomic"
+
+	"github.com/hanfei1991/microcosm/client"
+	"github.com/hanfei1991/microcosm/pb"
+)
+
+// ignoreAfterSuccClient will ignore the invocations after the first success,
+// to reduce unnecessary requests.
+// Currently it only implements PersistResource of MasterClient, calling other
+// method will panic.
+type ignoreAfterSuccClient struct {
```

Comment:
Why is it necessary? I think `PersistResource` should ideally be idempotent.

Reply:
Now on every file's Close worker will call `PersistResource`, that's too many RPCs if files are belong to one resource